### PR TITLE
docs: ZkUseFactoryDep cheatcode documentation

### PR DESCRIPTION
**Pull request by Jrigada**
- New page explaining the usage of the new cheatcode `ZkUseFactoryDep`
- Example of a multisig smart account deployment using `ZkUseFactoryDep`

Addresses #44

### CODE REVIEW

**Review by nbaztec — APPROVED** [review]

File: src/zksync-specifics/cheatcodes/zk-use-factory-dep.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+### Signature
+
+```solidity
+function zkUseFactoryDep(string calldata name) external pure;
+```
+
+### Description
+
+Enables marking a bytecode as a factory dependency for the next CREATE or CALL and cleared after.
```
Comment (Karrq) [inline src/zksync-specifics/cheatcodes/zk-use-factory-dep.md:11]:
```suggestion
Marks a given contract as a factory dependency only for the next CREATE or CALL, unmarking it afterwards, similar to [`prank`](../../cheatcodes/prank.md).
```

File: src/zksync-specifics/cheatcodes/zk-use-factory-dep.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+This cheatcode is useful when deploying contracts through factories, as it allows marking bytecode as a factory dependency and deploying contracts by passing their bytecode hash to the factory.
```
Comment (Karrq) [inline src/zksync-specifics/cheatcodes/zk-use-factory-dep.md:13]:
```suggestion
This cheatcode is useful when deploying contracts through factories that do not depend directly on a given contract, as it allows to explicitly mark this type of contract as a factory dependency, allowing the factory to deploy the contract.

More information on factory dependencies in the [official ZKsync docs](https://docs.zksync.io/build/developer-reference/ethereum-differences/contract-deployment#note-on-factory_deps).
```

File: src/zksync-specifics/examples/smart-account.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+
+        vm.stopBroadcast();
+    }
+}
+```
+
+### Running the script
+
+```sh
+zkforge script ./script/DeployMultisig.s.sol:DeployMultisig --rpc-url <RPC_URL> --private-key <PRIVATE_KEY> --broadcast --via-ir --system-mode true --zksync
```
Comment (Karrq) [inline src/zksync-specifics/examples/smart-account.md:405]: are we officially going for `zkforge`?

Comment (Karrq) [inline src/zksync-specifics/examples/smart-account.md:405]:
```suggestion
zkforge script ./script/DeployMultisig.s.sol:DeployMultisig --rpc-url <RPC_URL> --private-key <PRIVATE_KEY> --broadcast --via-ir --zk-enable-eravm-extensions --zksync
```

File: src/zksync-specifics/cheatcodes/zk-use-factory-dep.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+### Examples
+
+```solidity
+contract Deployer {
+    // Deploy the factory
+    Factory factory = new Factory(multisigBytecodeHash);
```
Comment (Karrq) [inline src/zksync-specifics/cheatcodes/zk-use-factory-dep.md:20]: Can we have a few extra lines to explain what's happening here? Specifically: 
* `Factory` does not depend on `TwoUserMultisig` directly
* We thus need to mark it explicitly

File: src/zksync-specifics/examples/smart-account.md [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+            factoryArtifact,
+            ".hash"
+        );
+        Create2Factory create2Factory = new Create2Factory();
+        address multisigAddress = create2Factory.create2(
+            salt,
+            factoryBytecodeHash,
+            abi.encode(owner1, owner2)
+        );
+        console.log("Multisig deployed at: ", multisigAddress);
```
Comment (Karrq) [inline src/zksync-specifics/examples/smart-account.md:395]: I don't understand this section - we just deployed `AAFactory` and even used it, but now we do the same? But with a different constructor even?
Above we have `bytecodeHash` and here it's `owner1,owner2` encoded...
Also technically speaking we can already make use of the `Create2Factory` implicitly with the CREATE2 syntax ;)

File: src/zksync-specifics/cheatcodes/zk-use-factory-dep.md (hunk 2 above)
Comment (Karrq) [reply src/zksync-specifics/cheatcodes/zk-use-factory-dep.md:13]: I like concise but for documentation we should be explicit on what problem this addresses, how it helps, and what you can do with it, outside of the examples later

**Review by Karrq — CHANGES_REQUESTED** [review]

File: src/zksync-specifics/examples/smart-account.md (hunk 3 above)
Comment (Jrigada) [reply src/zksync-specifics/examples/smart-account.md:405]: Good catch

**Review by tomimor — COMMENTED** [review]
💯

**Review by Karrq — APPROVED** [review]

**Pull request merged**